File: beginner/1006/average_2.c
#include <stdio.h>

int	main(void)
{
	double	weights[3];
	double	grades[3];
	double	grades_w;
	double	weight_sum;
	double	result;

	weights[0] = 2;
	weights[1] = 3;
	weights[2] = 5;
	scanf("%lf %lf %lf", &grades[0], &grades[1], &grades[2]);
	grades_w = grades[0] * weights[0] \
		+ grades[1] * weights[1] + grades[2] * weights[2];
	weight_sum = weights[0] + weights[1] + weights[2];
	result = grades_w / (double)weight_sum;
	printf("MEDIA = %.1lf\n", result);
	return (0);
}

File: beginner/1006/Average2.java
// challenge link - https://judge.beecrowd.com/en/problems/view/1006

import java.io.IOException;
import java.util.Scanner;

public class Average2 {
	public static void main(String[] args) {
		double[] weights = { 2, 3, 5 };
		double gradeSum = 0.0;
		double weightSum = 0.0;
		Scanner reader = new Scanner(System.in);

		for (int i = 0; i < weights.length; i++) {
			gradeSum += reader.nextDouble() * weights[i];
			weightSum += weights[i];
		}

		double average = gradeSum / weightSum;
		System.out.println("MEDIA = " + String.format("%.1f", average));
		reader.close();
	}
}
